core/rawdb/accessors_chain.go

```diff
@@ -253,6 +254,43 @@ func ReadTd(db DatabaseReader, hash common.Hash, number uint64) *big.Int {
 	return td
 }
 
+// ReadAccumulatedEpochUptime retrieves the so-far accumulated uptime array for the validators of the specified epoch
+func ReadAccumulatedEpochUptime(db DatabaseReader, epoch uint64) []istanbul.Uptime {
+	data, _ := db.Get(uptimeKey(epoch))
+	if len(data) == 0 {
+		log.Debug("uptime-trace: ReadAccumulatedEpochUptime = EMPTY", "epoch", epoch)
```

Comment:
nit: We could make these Traces or remove them altogether, had added them to get a sense of how the code executes during the E2E tests

```suggestion
		log.Trace("uptime-trace: ReadAccumulatedEpochUptime = EMPTY", "epoch", epoch)
```